spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/AdminApplication.java

```diff
@@ -17,12 +17,18 @@
 package org.springframework.cloud.data.admin;
 
 import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.yarn.boot.YarnAppmasterAutoConfiguration;
+import org.springframework.yarn.boot.YarnClientAutoConfiguration;
+import org.springframework.yarn.boot.YarnContainerAutoConfiguration;
 
 /**
  * @author Mark Fisher
  */
 @SpringBootApplication
+@EnableAutoConfiguration(exclude = { YarnClientAutoConfiguration.class, YarnAppmasterAutoConfiguration.class,
```

Comment:
There are boot auto-configuration classes in spring-yarn. `YarnAppmasterAutoConfiguration` and `YarnContainerAutoConfiguration` are expected to kick in when fat-jar is ran either as yarn `application master` or `container`. I excluded `YarnClientAutoConfiguration` to fix bean name clash which originated from component scan(read desc of this PR). I could have just changed `YarnConfiguration` class name which would have fixed bean name clash but felt it would have been a bit wrong. So ended up excluding all because those are not really needed in web context.

"admin on yarn", good point it would break because of same component scan issue. Component scan is evil and imho we should avoid it. Think we should have a broader discussion about it.